pkg/tls/cert.go

```
@@ -111,6 +111,51 @@ func CertKeyPairExists(name, dir string) (bool, error) {
 	return true, nil
 }
 
+// CertValid returns true if a matching certificate exist
+// Matching is defined as having the expected CN and SANs
+// A certificate with a wrong CN or that doesn't contain the expected SANs, will return a warning
+func CertValid(CN string, SANs []string, name, dir string) (valid bool, warn []error, err error) {
```

Got it. So the "return a single warning" option is not viable. I think doing complete validation over failing fast would result in a better UX. For the case where there are multiple issues with the certs, the user will have to run validation multiple times to discover each one.